src/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -0,0 +1,970 @@
+
+        /// <summary>
+        /// Contains any additional data provided by a derived type, typically set when calling <see cref="WriteTo(BinaryWriter, byte[])"/>.</param>
+        /// </summary>
+        protected virtual byte[] CustomSerializationData
+        {
+            get
+            {
+                return _userSerializationData;
+            }
+        }
+
+        /// <summary>
+        /// Allow the association of claims with this instance of <see cref="ClaimsIdentity"/>. 
+        /// The claims will not be serialized or added in Clone(). They will be included in searches, finds and returned from the call to <see cref="ClaimsIdentity.Claims"/>.
+        /// </summary>               
+        internal Collection<IEnumerable<Claim>> ExternalClaims
+        {
+            get { return _externalClaims; }
+        }
+
+        /// <summary>
+        /// Gets or sets the label for this <see cref="Identity"/>
+        /// </summary>
+        public string Label
+        {
+            get { return _label; }
+            set { _label = value; }
+        }
+
+        /// <summary>
+        /// Gets the Name of this <see cref="ClaimsIdentity"/>.
+        /// </summary>        
+        /// <remarks>Calls <see cref="FindFirst(string)"/> where string == NameClaimType, if found, returns <see cref="Claim.Value"/> otherwise null.</remarks>
+        public virtual string Name
+        {
+            // just an accessor for getting the name claim
+            get
+            {
+                Claim claim = FindFirst(_nameClaimType);
+                if (claim != null)
+                {
+                    return claim.Value;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the value that identifies 'Name' claims. This is used when returning the property <see cref="ClaimsIdentity.Name"/>.
+        /// </summary>
+        public string NameClaimType
+        {
+            get { return _nameClaimType; }
+        }
+
+        /// <summary>
+        /// Gets the value that identifies 'Role' claims. This is used when calling <see cref="ClaimsPrincial.IsInRole"/>.
+        /// </summary>
+        public string RoleClaimType
+        {
+            get { return _roleClaimType; }
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ClaimsIdentity"/> with values copied from this object.
+        /// </summary>
+        public virtual ClaimsIdentity Clone()
+        {
+            return new ClaimsIdentity(this);
+        }
+
+        /// <summary>
+        /// Adds a single <see cref="Claim"/> to an internal list.
+        /// </summary>
+        /// <param name="claim">the <see cref="Claim"/>add.</param>
+        /// <remarks>If <see cref="Claim.Subject"/> != this, then Claim.Clone(this) is called before the claim is added.</remarks>
+        /// <exception cref="ArgumentNullException">if 'claim' is null.</exception>
+        [SecurityCritical]
```

Comment:
@venkat-raman251 please go ahead and strip the Security* attributes when you do the ComVisible attributes.